parquet-variant/src/main/java/org/apache/parquet/variant/VariantUtil.java

```diff
@@ -0,0 +1,843 @@
+    }
+    int start = pos + 1 + U32_SIZE;
+    int length = readUnsigned(value, pos + 1, U32_SIZE);
+    checkIndex(start + length - 1, value.length);
+    return Arrays.copyOfRange(value, start, start + length);
+  }
+
+  public static String getString(byte[] value, int pos) {
+    checkIndex(pos, value.length);
+    int basicType = value[pos] & BASIC_TYPE_MASK;
+    int typeInfo = (value[pos] >> BASIC_TYPE_BITS) & PRIMITIVE_TYPE_MASK;
+    if (basicType == SHORT_STR || (basicType == PRIMITIVE && typeInfo == LONG_STR)) {
+      int start;
+      int length;
+      if (basicType == SHORT_STR) {
+        start = pos + 1;
+        length = typeInfo;
+      } else {
+        start = pos + 1 + U32_SIZE;
+        length = readUnsigned(value, pos + 1, U32_SIZE);
+      }
+      checkIndex(start + length - 1, value.length);
+      return new String(value, start, length);
+    }
+    throw unexpectedType(Type.STRING);
+  }
+
+  public static java.util.UUID getUUID(byte[] value, int pos) {
+    checkIndex(pos, value.length);
+    int basicType = value[pos] & BASIC_TYPE_MASK;
+    int typeInfo = (value[pos] >> BASIC_TYPE_BITS) & PRIMITIVE_TYPE_MASK;
+    if (basicType != PRIMITIVE || typeInfo != UUID) {
+      throw unexpectedType(Type.UUID);
+    }
+    int start = pos + 1;
+    checkIndex(start + UUID_SIZE - 1, value.length);
+    ByteBuffer bb = ByteBuffer.wrap(value, start, UUID_SIZE).order(ByteOrder.BIG_ENDIAN);
+    return new java.util.UUID(bb.getLong(), bb.getLong());
+  }
+
+  /**
+   * A helper class representing the details of a Variant object, used for `ObjectHandler`.
+   */
+  public static class ObjectInfo {
+    /** Number of object fields. */
+    public final int numElements;
+    /** The integer size of the field id list. */
+    public final int idSize;
+    /** The integer size of the offset list. */
+    public final int offsetSize;
+    /** The starting index of the field id list in the variant value array. */
+    public final int idStart;
+    /** The starting index of the offset list in the variant value array. */
+    public final int offsetStart;
+    /** The starting index of field data in the variant value array. */
+    public final int dataStart;
+
+    public ObjectInfo(int numElements, int idSize, int offsetSize, int idStart, int offsetStart, int dataStart) {
+      this.numElements = numElements;
+      this.idSize = idSize;
+      this.offsetSize = offsetSize;
+      this.idStart = idStart;
+      this.offsetStart = offsetStart;
+      this.dataStart = dataStart;
+    }
+  }
+
+  /**
+   * An interface for the Variant object handler.
+   * @param <T> The return type of the handler
+   */
+  public interface ObjectHandler<T> {
+    /**
+     * @param objectInfo The details of the Variant object
+     */
+    T apply(ObjectInfo objectInfo);
+  }
+
+  /**
+   * An interface for the Variant object handler.
+   * @param <T> The return type of the handler
+   */
+  public interface ObjectHandlerException<T> {
```

Comment:
Is there a better solution than this? It's not ideal that this library accounts for a specific use of `ObjectHandler` by creating a secondary handler that passes an IOException through. I would prefer using a `getObjectInfo(byte[], int)` for those cases instead of adding new handlers.